application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadAutoArchiver.java

```diff
@@ -122,18 +127,44 @@ private static boolean shouldBeArchived(MessageChannel channel, Instant archiveA
     }
 
     private void handleArchiveFlow(ThreadChannel threadChannel, MessageEmbed embed) {
-        Consumer<Throwable> handleFailure = error -> {
+
+        Supplier<RestAction<Void>> archiveThread =
+                () -> threadChannel.getManager().setArchived(true);
+
+        Function<Throwable, RestAction<Void>> handleFailure = error -> {
             if (error instanceof ErrorResponseException) {
-                logger.warn("Unknown error occurred during help thread auto archive routine",
+                logger.warn(
+                        "Unknown error occurred during help thread auto archive routine, archiving thread",
                         error);
             }
+            return archiveThread.get();
         };
 
+        Function<Result<Member>, RestAction<Message>> sendEmbedWithMention =
+                member -> threadChannel.sendMessage(member.get().getAsMention()).addEmbeds(embed);
+
+        Function<Result<Member>, RestAction<Message>> sendEmbedWithoutMention =
+                member -> threadChannel.sendMessageEmbeds(embed);
+
         threadChannel.getGuild()
             .retrieveMemberById(threadChannel.getOwnerIdLong())
-            .flatMap(author -> threadChannel.sendMessage(author.getAsMention()).addEmbeds(embed))
-            .flatMap(any -> threadChannel.getManager().setArchived(true))
-            .queue(any -> {
-            }, handleFailure);
+            .mapToResult()
+            .flatMap(member -> {
+                if (member.isSuccess()) {
+                    return sendEmbedWithMention.apply(member);
+                }
+
+                logger.debug("Unable to mention user", member.getFailure());
+                return sendEmbedWithoutMention.apply(member);
+            })
+            .mapToResult()
+            .flatMap(sentEmbed -> {
+                if (sentEmbed.isSuccess()) {
+                    return archiveThread.get();
+                }
+
+                return handleFailure.apply(sentEmbed.getFailure());
```

Comment:
```suggestion
                if (sentEmbed.getFailure() instanceof ErrorResponseException) {
                    // log
                }

                return threadChannel.getManager().setArchived(true);
```

Reply:
what's the issue with earlier implementation? It tells a beautiful story, if embed is a success archive the thread otherwise handle failure..which logs and then regardless archives it. It looks like you're trying to squeeze in the failure part at the cost of readability.

Reply:
if its failure log, and archive anyways. doesn't need to be more complex than that

Reply:
and it's doing that exactly, what im trynna say is you're squeezing in the failure part. Im trying to clean the chain cleaner, like i said it's already doing too much.

Reply:
The functional implementation is local to method, so you dont have to jump around. It has a name as well, which abstracts away the details when reading pipeline. So something bad happened, oki it's calling a function `handleFailure` which consumes a `throwable` so maybe ill look at later but ik it handles failures. 

I don't wanna do the if check inline because, it just adds more brackets then you're keeping track of what's what inside chain. Ill do the changes but i still feel the same way.